music_assistant/providers/airplay/player.py

```diff
@@ -202,16 +232,69 @@ async def pause(self) -> None:
+            if not self.stream or not self.stream.running:
                 return
-            await self.raop_stream.send_cli_command("ACTION=PAUSE")
+            await self.stream.send_cli_command("ACTION=PAUSE")
+
+    async def ap_version_consistency(self) -> None:
+        """Ensure consistency between configured airplay version and mdns discovery info."""
+        # Configured version takes precedence
+        configured_version: int = await self.mass.config.get_player_config_value(
+            self.player_id, CONF_AIRPLAY_VERSION
+        )  # type: ignore[assignment]
+        if configured_version != self._airplay_version:
+            self.logger.info(
+                f"{self.name}:configured AirPlay version ({configured_version}) does not match "
+                f"detected device AirPlay version ({self._airplay_version}). "
+                f"Updating to {configured_version}."
+            )
+            # update to the configured version and refresh discovery info
+            info: AsyncServiceInfo | None = None
+            if configured_version == 1:
+                type_ = "_raop._tcp.local."
+                # RAOP discovery name is in format X@Y._raop._tcp.local.
+                # where X is the capitalized deviceid with colons stripped
+                # and Y is the device name
+                deviceid = str(self.discovery_info.decoded_properties.get("deviceid"))
+                name = (
+                    deviceid.upper().replace(":", "")
+                    + "@"
+                    + self.discovery_info.name.replace("_airplay", "_raop")
+                )
+            elif configured_version == 2:
+                # AirPlay discovery name is in format Y._airplay._tcp.local.
+                type_ = "_airplay._tcp.local."
+                name = self.discovery_info.name.split("@")[-1].replace("_raop", "_airplay")
+            else:  # guard against invalid configured version
+                self.logger.warning(
+                    f"{self.name}:unsupported AirPlay version configured: {configured_version}"
+                )
+                return
+            info = AsyncServiceInfo(type_, name)
+            if await info.async_request(self.mass.aiozc.zeroconf, 3000):
+                self.set_discovery_info(info, str(self.name))
+                self._airplay_version = configured_version
+            else:
+                # This will happen when the device does not support the configured version
+                # in which case, we just stick with the discovered version
+                # Should we force an update of the config entry to match the discovered version??
+                self.logger.error(
+                    f"{self.name}:failed to refresh discovery info for AirPlay version "
+                    f"{configured_version}. "
+                    f"Continuing with previously detected version {self._airplay_version}."
+                )
```

Comment:
I am not sure what we should do in this instance. I don't think it is possible for there to be zero discovery information, so this case should never happen. I would appreciate advice on what to do please.